AestraUI/Base/NUIScrollbar.cpp

```
@@ -72,6 +72,35 @@ void NUIScrollbar::onRender(NUIRenderer& renderer)
+void NUIScrollbar::onUpdate(double deltaTime)
+{
+    if (isAnimating_ && !isDragging_) {
+        animationTime_ += deltaTime;
+        const double t = std::clamp(animationTime_ / animationDuration_, 0.0, 1.0);
+        const double eased = 1.0 - std::pow(1.0 - t, 3.0);
+        const double next = animationStartValue_ + (animationTargetValue_ - animationStartValue_) * eased;
+        setCurrentRange(next, currentRangeSize_);
+        triggerScroll();
+
+        if (t >= 1.0) {
+            isAnimating_ = false;
```

Cppcheck (2.20.0)

[style] The function 'getThumbSize' is never used.

(unusedFunction)